examples/resnet50/rocal.cpp

```diff
@@ -0,0 +1,246 @@
+  // before sending the batch on
+  std::string endpoint = client->workerLoad("rocal", parameters);
+  amdinfer::waitUntilModelReady(client, endpoint);
+  // -load
+  return endpoint;
+}
+
+/**
+ * @brief The command-line arguments are parsed in two phases. There's the
+ * common arguments that are initialized by parseArgs that are shared by all the
+ * C++ examples in this directory and then example-specific settings are
+ * initialized here
+ *
+ * @param argc number of arguments
+ * @param argv arguments
+ * @return Args
+ */
+Args getArgs(int argc, char** argv) {
+  Args args = parseArgs(argc, argv);
+
+  // if the model is not specified and no endpoint flag is passed, attempt to
+  // set a default value for the model to use for loading
+  if (args.path_to_model.empty() && args.endpoint.empty()) {
+    const auto* root_str = std::getenv("AMDINFER_ROOT");
+    assert(root_str != nullptr);
+    fs::path root{root_str};
+    args.path_to_model = root / "examples/resnet50/rocal_pipeline.json";
+    args.path_to_image = root / "tests/assets/imagenet-dog.jpg";
+  }
+  return args;
+}
+
+int main(int argc, char* argv[]) {
+  try {
+    std::cout << "Running the rocAL example in C++\n";
+
+    Args args = getArgs(argc, argv);
+
+    const auto http_port_str = std::to_string(args.http_port);
+    const auto server_addr = "http://" + args.ip + ":" + http_port_str;
+    amdinfer::HttpClient client{server_addr};
+
+    std::optional<amdinfer::Server> server;
+    if (args.wait) {
+      // if wait is true, skip ahead to waiting for the server to become ready
+    } else if (args.ip == "127.0.0.1" && !client.serverLive()) {
+      std::cout << "No server detected. Starting locally...\n";
+      server.emplace();
+      server.value().startHttp(args.http_port);
+    } else if (!client.serverLive()) {
+      std::cerr << "Could not connect to server at " << server_addr << "\n";
+      return 1;
+    } else {
+      // the server is reachable so continue on
+    }
+
+    std::cout << "Waiting until the server is ready...\n";
+    amdinfer::waitUntilServerReady(&client);
+
+    std::string endpoint;
+    if (!args.endpoint.empty()) {
+      endpoint = args.endpoint;
+      if (!client.modelReady(endpoint)) {
+        std::cerr << "Model at " << endpoint
+                  << " does not exist or isn't ready. Verify the endpoint or "
+                     "omit the --endpoint flag to load a new worker\n";
+        return 1;
+      }
+    } else {
+      std::cout << "Loading worker...\n";
+      endpoint = load(&client, args);
+    }
+
+    std::vector<std::string> paths = resolveImagePaths(args.path_to_image);
+
+    Images imageBuffer = loadImages(paths);
+    std::vector<amdinfer::InferenceRequest> requests =
+      constructRequests(imageBuffer);
+    assert(paths.size() == requests.size());
+    const auto num_requests = requests.size();
+
+    std::cout << "Preprocessing image...\n";
```

Comment:
is this supposed to be there in the production code?

Reply:
Same comment is available on migraphx example. 
I assume it is okay since it is an example app.